op-dispute-mon/mon/status.go

```diff
@@ -0,0 +1,28 @@
+package mon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/types"
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching"
+)
+
+type statusLoader struct {
+	caller *batching.MultiCaller
+}
+
+func NewStatusLoader(caller *batching.MultiCaller) *statusLoader {
+	return &statusLoader{caller: caller}
+}
+
+func (s *statusLoader) GetStatus(ctx context.Context, proxy common.Address) (types.GameStatus, error) {
+	fdg, err := contracts.NewFaultDisputeGameContract(proxy, s.caller)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create FaultDisputeGameContract: %w", err)
+	}
```

Comment:
We should ensure this is given the full `GameMetaData` so that it can create the correct contract bindings based on the game type.  For now we only have the one type but this is the key point we can leverage to support multiple types.  It should return an error if the game type is unknown as well since we don't want to silently monitor unexpected game types and potentially miss details about them.

Reply:
Actually I think I'd change this struct to be more of a factory for the game contract, then the monitoring can just be given the contract (rather than having an additional layer wrapping that we'd have to add new methods to for everything we want to get out of the contract).  So something like:

```
type MonitorableGame interface {
  GetStatus(ctx context.Context) ...
}

type ContractLoader struct {
    caller *batching.MultiCaller
}

func (l *ContractLoader) CreateContract(game types.GameMetadata) MonitorableGame {
    switch game.GameType {
        case alphabet, cannon:
            // create binding and return
        default:
           return nil, ErrUnknownGameType
    }
}
```